source/common/filter/config_discovery_impl.cc

```diff
@@ -131,6 +131,9 @@ void FilterConfigSubscription::onConfigUpdate(
   last_type_url_ = type_url;
   last_version_info_ = version_info;
   last_factory_name_ = factory_name;
+  filter_config_provider_manager_.storeFilterConfigs(filter_config, last_type_url_,
```

Comment:
Could we avoid duplicating `filter_config`? I think `message` is the same content?

Reply:
Just to add more context: I think there is one subscription on the main thread, so we can probably reuse that one for the config dump. That helps reducing overhead with more copies of the config.

Reply:
Yes, that's a great idea! The latest code is directly to dump the data in the already existing subscriptions_ hash map: https://github.com/envoyproxy/envoy/blob/b0d79665240cdfe3d87280283d28ac30334542e9/source/common/filter/config_discovery_impl.h#L392

With this, there is no new memory consumption added by this PR.